docs/rfc/rfc-19-random-beacon-v2.adoc

```diff
@@ -0,0 +1,225 @@
+
+The last release of the random beacon got its genesis Sep 16th 2020 and worked
+until November 11th when it was stopped with a panic button as a result of an
+unannounced https://github.com/keep-network/keep-core/blob/main/docs/status-reports/2020-11-11-retro-geth-hardfork.adoc[Ethereum
+hard fork].
+
+For the time it was working, the beacon produced more than 1700 relay entries
+and 70 groups.
+
+Both on-chain and off-chain components proved to be successful but - just like
+with any other product - some pain points were identified by the users.
+The most problematic aspects were the construction of pricing scheme and too
+aggressive timeouts. This RFC aims at describing these problems and proposing
+changes that should make the beacon easier to operate and more consistent with
+tBTC v2.
+
+=== ETH-based pricing
+
+Having reimbursements for various on-chain actions in ETH was a nice feature but
+also complex in practice. Highly volatile gas prices on mainnet made the gas price
+ceiling value impossible to maintain up-to-date. Given that the gas price
+ceiling had to be higher than the current gas price and that the fee for future
+group creation was calculated based on the gas price ceiling, new groups were 
+getting created too often. 
+
+With too many groups being created, operators were spending too much ETH on
+ticket submission and too low rewards were getting accumulated per group making
+the reward withdrawal operation not profitable enough or not profitable at all
+in the case of small stakers.
+
+Extra submitter rewards added on top of reimbursements were hard to understand
+for staking providers and calculating the provider’s fee was complicated given
+the non-reimbursable cost of ticket submission. Surprisingly, it is much easier
+for staking providers to calculate their fee and costs in tBTC v1, having a
+clean rule that no operations are reimbursed and the only reward they receive
+is KEEP plus TBTC fees.
+
+=== Aggressive timeouts
+
+Relay entry timeouts were pretty aggressive. Set to 64x6 = 384 blocks, they were
+giving about 1.5 hours to submit a relay entry before the minimum stake from each
+group member was slashed. This timeout proved to be not enough to notify,
+investigate, and fix a problem. Three groups were slashed in a row during the
+November 11th incident that begun around 08:00 UTC, before any US-based
+operators noticed the problem.
+
+
+== Proposal
+
+=== Group creation
+
+New groups should be created with a fixed, governable frequency of relay
+requests. Instead of a ticket-based approach for group selection, we should
+use sortition pools. Group creation start transaction should be embedded into
+relay request transaction and should lock a sortition pool. From this moment,
+no operator can enter or leave the pool. All off-chain clients should perform
+group selection optimistically, using a view sortition function call and perform
+DKG. Once the group submits the result to the chain, a challenge period starts.
```

Comment:
we should note that the sortition is based on the entry whose request locked the sortition pool

Reply:
Good point, the current version is not clear about it at all. I'll improve it.

Reply:
Please see https://github.com/keep-network/keep-core/pull/2585/commits/0c97106900a6469bcb4e2be2d7927fed24398f54.